pkg/networking/networking.go

```diff
@@ -61,45 +61,35 @@ func (crt *customRoundTripper) RoundTrip(request *http.Request) (*http.Response,
 func NewNetworkAccess(config configuration.Configuration) NetworkAccess {
 	// prepare logger
 	logger := log.New(os.Stderr, "NetworkAccess - ", config.GetInt(configuration.DEBUG_FORMAT))
-	if config.GetBool(configuration.DEBUG) == false {
+	if !config.GetBool(configuration.DEBUG) {
 		logger.SetOutput(io.Discard)
 	}
 
-	c := networkImpl{
+	return &networkImpl{
 		config:       config,
 		staticHeader: http.Header{},
 		logger:       logger,
 		proxy:        http.ProxyFromEnvironment,
 	}
-
-	return &c
 }
 
-func (n *networkImpl) AddHeaderField(key string, value string) {
-	n.staticHeader[key] = append(n.staticHeader[key], value)
+func (n *networkImpl) AddHeaderField(key, value string) {
+	n.staticHeader.Add(key, value)
```

Comment:
Safer to use the `http.Header` API as it takes care of converting keys to their canonical form.